db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -3546,4 +3614,129 @@ void tagsAreNotShuffledDuringMerge() throws SQLException {
       assertEquals(case8Tags, tagsHelper.getTagsOnActivity(case8Id, planId));
     }
   }
+
+  @Nested
+  class PlanMigrationTests{
+    /**
+     * Check that migration fails if:
+     *  (a) _plan_id is invalid and _new_model_id is valid
+     *  (b) _plan_id is valid and _new_model_id is invalid
+     *  (c) _plan_id and _new_model_id are both invalid
+     *
+     */
+    @Test
+    void checkFailsForNonexistentPlansOrModels() throws SQLException {
+      final int existingPlanId = merlinHelper.insertPlan(missionModelId);
+
+      try{
+        migratePlanToModel(-1, missionModelId);
+        fail();
+      }
+      catch (SQLException sqEx){
+        if(!sqEx.getMessage().contains("Plan -1 does not exist"))
+          throw sqEx;
+      }
```

Comment:
Please use `assertThrows` to model this behavior. This applies to every location of this pattern. The `try-catch-fail` pattern is vulnerable to forgetting to add the `fail` statement (which has occurred in the past).

```suggestion
final var sqEx = assertThrows(SQLException.class, () -> migratePlanToModel(-1, missionModelId));
if(!sqEx.getMessage().contains("Plan -1 does not exist")) {
    throw sqEx;
}
```

Reply:
fixed